### DISCUSSION TITLE
Can I insert my system date in a prompt?

### DISCUSSION THREAD

**Discussion by rob215x** [open-webui/open-webui#2796] [General]
My LLM "Lumina" doesn't know what day or time it is, unless I tell her. I can say, "Hey Lumina today is June 3, 2024 and it's 6:30pm. Do you know what day of the week it is?" and she'll give the correct answer. 

But I'd like to have a prompt that automatically inserts today's date and time so that when I start a new chat, Lumina already knows what day it is. Can I do that?

This would also be an example of how to insert any system vars into prompts! I may have missed it, but I couldn't find it in the documentation. 

Thanks!

**Comment by justinh-rahb** [comment]
`{{CURRENT_DATE}}`

**Comment by silentoplayz** [comment]
As of v0.2.0, two new variables were added: `{{CURRENT_DATE}}` and `{{USER_NAME}}` for dynamic prompts.

**Comment by rob215x** [comment]
Thanks!! Maybe I'm missing something obvious but its not working for me...

![Screenshot_2024-06-04_12-49-37](https://github.com/open-webui/open-webui/assets/20937625/16cd79e1-9737-40fd-abcf-0309bee4041f)

**Reply by justinh-rahb** [reply]
It's not intended to work that way. These tags are only replaced when used in system prompts, not user messages.

**Reply by rob215x** [reply]
Thanks for the quick response! So I need to use {{CURRENT_DATE}} inside my  SYSTEM """ ... """ text?

I just tried it inside the SYSTEM content and it doesn't seem to work there either.

**Reply by justinh-rahb** [reply]
Which version of WebUI are you using?

**Reply by rob215x** [reply]
Ahhhh, I'm on 0.1.125!! Let me update and try again. I'm really sorry, I used to get a little notification when there was an update. brb

**Comment by rob215x** [comment]
Ok, I updated!! I tried doing this:

![Screenshot_2024-06-04_13-26-33](https://github.com/open-webui/open-webui/assets/20937625/1838086a-6512-4f64-980b-8c761826bec9)

but when I type /date it still just outputs {{CURRENT_DATE}} instead of the actual date

I tried adding this to SYSTEM:
today's date is {{CURRENT_DATE}}

But I just get:

![Screenshot_2024-06-04_13-30-53](https://github.com/open-webui/open-webui/assets/20937625/0ad61be2-cb16-4075-baea-1adcaa3bf838)

**Reply by tjbck** [reply]
![image](https://github.com/open-webui/open-webui/assets/25473318/9278a135-f809-4d14-ae6d-895da57b2ad2)

**Reply by rob215x** [reply]
That worked!! Thanks

![Screenshot_2024-06-04_13-36-11](https://github.com/open-webui/open-webui/assets/20937625/4f62387c-d79f-4f96-beb0-53e55a32f086)

**Comment by rob215x** [comment]
@tjbck where can I read more about the difference between the general system prompt and the system prompt for individual models/modelfiles? I was expecting to put {{CURRENT_DATE}} in my modelfile. Thanks!!

By the way, I'm really enjoying open webui! I think it can be a great learning/teaching tool

**Comment by rob215x** [comment]
Okay, after a restart and a new chat, Lumina no longer knows the date. She thinks its 2022 and she has no idea what time it is. 

I'm just pointing this out because either I'm not sure how to use {{CURRENT_DATE}} correctly, or it's not working as intended. 
- Does {{CURRENT_DATE}} also contain the local time on my system?
- In the system prompt, should it say: The current date is {{CURRENT_DATE}} instead of just the var by itself?
- Why wouldn't it go in Workspace > Edit Model > System Prompt instead of General > System Prompt?

**Comment by megvadulthangya** [comment]
Hi guys! Im using ollama and openwebui both in one docker compose file... this current date thingy is not working to me... im on latest version... Any thoughs please?

**Reply by silentoplayz** [reply]
Working fine for me still (other than it thinking today's date is actually *tomorrow's* date).
![image](https://github.com/user-attachments/assets/9a7b5e20-73e5-4319-a625-e965f0ceee99)
![image](https://github.com/user-attachments/assets/3a4038cd-2b34-41b2-89c3-5ecc9b24137f)
